index.rst

```diff
@@ -10,16 +10,200 @@
+
+.. _general-priorities:
+
+General Priorities
+==================
+
+Of the outputs produced by the Alert Production pipeline, the most important to have correct and uncorrupted is the APDB, which serves as the source of truth both for later processing and for the externally available Prompt Products database.
+The alert stream is a close second, since it is the primary user interface for transient follow-up and cannot be "fixed" after the fact.
+The central repository and metrics are less important, because they are (almost) exclusively for internal use.
+
+As far as possible, the Prompt Processing pipeline should be idempotent -- in other words, multiple tries of the same pipeline on the same dataset should yield the same results, no matter what the state of the system.
+This is possible when rebroadcasting the alert stream -- in the worst case, duplicate alerts exist but have the same position and time -- and when running the pipeline from ISR through image differencing.
+However, for reasons explored in :ref:`association`, idempotency cannot be safely achieved with source association.
+
+.. _retries:
+
+Retrying Processing
+===================
+
+The execution framework can automatically recover from failures, in the sense of restoring full processing capacity without compromising the framework's state.
+However, recovering the data processing itself is the responsibility of the code running on each worker.
+
+Visit processing can fail at any point, including within and between tasks, and while performing external I/O.
+Even within a single worker, the order of events is not fully deterministic (e.g., two snaps could be ingested and processed up to snap combination in parallel).
+However, the possible failures can be divided into a handful of cases:
+
+- Any failure before the end of APDB writing (which is atomic as of May 2023) has no consequences beyond that worker's local repository.
+  The repository may have ingested raws, or partial pipeline outputs.
+- A failure after APDB writing but before the end of ``DiaPipelineTask``, which contains it, is recoverable only if the task can account for the fact that a previous session already wrote to the APDB.
+- A failure after APDB writing but before or during alert distribution cannot be aborted without leaving the APDB inconsistent with the alert stream.
+  However, retrying is possible, and easy if duplicate alerts are acceptable (see :ref:`general-priorities`).
+- A failure after alert distribution but but before or during central repo sync may leave the latter missing calexps and difference images, which are supposed to be exported to the Rubin Science Platform.
+- A failure after central repo sync may interfere with internal metrics and monitoring, but has no consequences for science users.
+
+In general, the Rubin Observatory project has followed a policy of processing data at most once, not at least once.
```

Comment:
I'm not sure this is always the case, but for AP in particular, dropping a few visits can still allow the 98% requirement to be met, so we indeed have erred on the side of "at most once".